### PR TITLE
[RUN-5628] Route WebContents events over to BrowserView

### PR DESCRIPTION
**Pull request by MichaelMCoates**
#### Description of Change

This PR Adds the following events from WebContents over to BrowserView:
- page-title-updated (also added to Window)
- did-change-theme-color (also added to Window)
- page-favicon-updated (also added to Window)
- crashed

Tests:
https://testing-dashboard.openfin.co/#/app/sessions/api/completed/5d8a81a67ba87401eaf566a8

Corresponding Core PR:
https://github.com/HadoukenIO/core/pull/968

#### Checklist

- [X] PR description included and stakeholders cc'd
- [X] `npm test` passes
- [X] automated tests are [changed or added](https://testing-dashboard.openfin.co/#/app/dashboard)
- [X] Integration tests added in this repo and in the [test dashboard](https://testing-dashboard.openfin.co/#/app/dashboard)
- [X] relevant documentation is changed or added
- [X] PR title starts with the JIRA ticket [pull request process](https://github.com/openfin/Internal-Wiki/wiki/Pull-Request-Process)
- [X] PR release notes describe the change in a way relevant to app-developers
- [X] PR has assigned reviewers

#### Release Notes

Notes: 
N/A, experimental

### CODE REVIEW

**Review by pbaize — APPROVED** [review]

**Review by pbaize — CHANGES_REQUESTED** [review]
You shouldn't have to add these to WindowEventMapping, but you should add them to the propagated maps

**Comment by MichaelMCoates** [comment]
> You shouldn't have to add these to WindowEventMapping, but you should add them to the propagated maps

Ahh, that's cool, changed.

**Review by rdepena — DISMISSED** [review]

File: tutorials/BrowserView.EventEmitter.md [hunk 1]
```diff
@@ -72,54 +72,91 @@ view.removeAllListeners("closed");
...
 
-#### attached
-Generated when a BrowserView attaches to a window. This event will fire during creation of a BrowserView. In that case, `previousTarget` identity will be the same as `target` identity.
+#### crashed
+Generated when a view has crashed.
+```js
+//This response has the following shape:
+{
+    name: "windowOne", //the name of the window.
+    reason: "killed", //possible values:
```
Comment (licui3936) [inline tutorials/BrowserView.EventEmitter.md:107]: name is  browserView name, not window name?

File: tutorials/BrowserView.EventEmitter.md [hunk 2]
```diff
@@ -72,54 +72,91 @@ view.removeAllListeners("closed");
...
 }
 ```
 
+#### did-change-theme-color
+Emitted when a page's theme color changes. This is usually due to encountering a meta tag: <meta name='theme-color' content='#ff0000'>
+```js
+{
+    color: "#FF0000",
+    name: "windowOne", //the name of the window.
+    topic: "view",
```
Comment (licui3936) [inline tutorials/BrowserView.EventEmitter.md]: same here

File: tutorials/BrowserView.EventEmitter.md [hunk 3]
```diff
@@ -72,54 +72,91 @@ view.removeAllListeners("closed");
...
     uuid: "454C7F31-A915-4EA2-83F2-CFA655453C52" // the uuid of the BrowserView
 }
 ```
 
+#### did-change-theme-color
+Emitted when a page's theme color changes. This is usually due to encountering a meta tag: <meta name='theme-color' content='#ff0000'>
+```js
+{
+    color: "#FF0000",
+    name: "windowOne", //the name of the window.
```
Comment (licui3936) [inline tutorials/BrowserView.EventEmitter.md]: same here

File: tutorials/BrowserView.EventEmitter.md [hunk 4]
```diff
@@ -133,15 +170,42 @@ Generated when a BrowserView is hidden.
...
-#### destroyed
-Generated when a BrowserView is destroyed.
+#### page-favicon-updated
+Emitted when page receives favicon urls.
+```js
+{
+    favicons: [
+        "http://www.openfin.co/favicon.ico"
+    ],
+    name: "windowOne", //the name of the window.
```
Comment (licui3936) [inline tutorials/BrowserView.EventEmitter.md]: same here

File: tutorials/BrowserView.EventEmitter.md [hunk 5]
```diff
@@ -133,15 +170,42 @@ Generated when a BrowserView is hidden.
...
+    uuid: "AppUUID" //(string) the UUID of the application the window belongs to.
+}
+```
+
+#### page-title-updated
+Fired when page title is set during navigation. explicitSet is false when title is synthesized from file url.
+```js
+{
+    explicitSet: true, // false when title is synthesized from file url.
+    name: "windowOne", //the name of the window.
```
Comment (licui3936) [inline tutorials/BrowserView.EventEmitter.md]: same here

File: tutorials/BrowserView.EventEmitter.md (hunk 1 above)
Comment (MichaelMCoates) [reply tutorials/BrowserView.EventEmitter.md:107]: Updated!

**Comment by MichaelMCoates** [comment]
@pbaize any more changes?

File: src/api/events/browserview.ts [hunk 6]
```diff
@@ -3,6 +3,7 @@ import { WindowEvent } from './base';
 
 export interface ViewEventMapping<Topic = string, Type = string> extends WebContentsEventMapping {
     'attached': WindowEvent<Topic, Type>;
+    'crashed': WindowEvent<Topic, Type>;
```
Comment (pbaize) [inline src/api/events/browserview.ts]: Shouldn't this go into webcontents?

File: src/api/events/webcontents.ts [hunk 7]
```diff
@@ -21,4 +21,7 @@ export interface WindowResourceResponseReceivedEvent<Topic, Type> extends Window
 export interface WebContentsEventMapping<Topic = string, Type = string> extends BaseEventMap {
     'resource-load-failed': WindowResourceLoadFailedEvent<Topic, Type>;
     'resource-response-received': WindowResourceResponseReceivedEvent<Topic, Type>;
+    'did-change-theme-color': WindowEvent<Topic, Type>;
```
Comment (pbaize) [inline src/api/events/webcontents.ts:25]: Do you need to add these to Window's and View's propagated maps?

Comment (MichaelMCoates) [reply src/api/events/webcontents.ts]: I'm handling propagation in a separate PR once all the PRs are in.

File: src/api/events/browserview.ts (hunk 6 above)
Comment (MichaelMCoates) [reply src/api/events/browserview.ts]: Right, good catch, changing now.

File: src/api/events/webcontents.ts (hunk 7 above)
Comment (MichaelMCoates) [reply src/api/events/webcontents.ts:25]: I'm handling Propagation in a separate PR once all PRs are in.

**Review by pbaize — APPROVED** [review]

**Review by datamadic — APPROVED** [review]

**Pull request merged**